### PR TITLE
Execute commands using user's shell.

### PR DESCRIPTION
**Pull request by russjones**
**Purpose**

As mentioned in https://github.com/gravitational/teleport/issues/923 we were not parsing commands correctly. Upon further investigation to maintain interoperability with OpenSSH we should not parse commands and instead call the users shell with `-c` like OpenSSH does.

**Implementation**

* In `exec.go` instead of trying to parse the command we pass it as-is to the user shell.
* The bulk of this PR is test coverage for a variety of scenarios to prevent regressions.

**Related Issues**

Fixes https://github.com/gravitational/teleport/issues/923

### CODE REVIEW

**Comment by russjones** [comment]
@gravitational-jenkins test this please

**Comment by russjones** [comment]
@gravitational-jenkins test this please

**Review by klizhentas — APPROVED** [review]

File: lib/srv/exec.go [hunk 1]
```diff
@@ -137,9 +137,11 @@ func prepInteractiveCommand(ctx *ctx) (*exec.Cmd, error) {
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	// this configures shell to run in 'login' mode
+	// this configures shell to run in 'login' mode. from the bash man page:
```
Comment (klizhentas) [inline lib/srv/exec.go]: you are mentioning bash docs here, what about other shells? what does OpenSSH do in case if shell is not supported?

**Comment by klizhentas** [comment]
actually, I have one question about shell compatibility

File: lib/srv/exec.go (hunk 1 above)
Comment (russjones) [inline lib/srv/exec.go]: Good catch, rolled back this change, looks OpenSSH sets the first argument to `-{shell name}`: https://github.com/openssh/openssh-portable/blob/master/session.c#L1650-L1663

**Review by klizhentas — APPROVED** [review]
ok, you are good to go then

**Comment by kontsevoy** [comment]
LGTM

**Review by kontsevoy — APPROVED** [review]

**Pull request merged**